core/dbt/parser/schemas.py

```diff
@@ -1027,14 +1084,33 @@ def __init__(self, schema_parser: SchemaParser, yaml: YamlBlock):
         self.schema_parser = schema_parser
         self.yaml = yaml
 
-    def parse_metric(self, unparsed: UnparsedMetric) -> ParsedMetric:
+    def parse_metric(self, unparsed: UnparsedMetric) -> None:
```

Comment:
This seems wrong but if the metric is disabled there's nothing to return.  Or should I be building a metric patch?

Reply:
The whole 'patch' thing is intended for nodes that are built from other files (such as SQL files) but that have config or other attributes changes in a schema file. 

I think what we should do is change the way that 'parse' works for metrics and exposures. The way this is working right now the node is added to disabled in the 'parse_metric' and 'parse_exposure' methods, but not added to the manifest until the node is returned in the 'read_files' method. I think we should stop returning anything from parse for metrics and exposures and move adding the node to the manifest into 'parse_metric' and 'parse_exposure'. If you look at the various parser returns in 'read_files' we already have a number of parsers which don't do anything with the 'parse' method return. Mostly it's used for handling tests.